src/main/java/com/indeed/authorization/client/validators/IndeedAccessTokenClaimsVerifier.java

```diff
@@ -0,0 +1,139 @@
+package com.indeed.authorization.client.validators;
+
+import com.indeed.authorization.client.claims.IndeedAccessTokenClaimsSet;
+import com.nimbusds.jose.proc.JWKSecurityContext;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.proc.BadJWTException;
+import com.nimbusds.jwt.proc.ClockSkewAware;
+import com.nimbusds.jwt.proc.JWTClaimsSetVerifier;
+import com.nimbusds.oauth2.sdk.ParseException;
+import com.nimbusds.oauth2.sdk.id.ClientID;
+import com.nimbusds.oauth2.sdk.id.Issuer;
+
+import java.util.Arrays;
+import java.util.HashSet;
+
+import static com.indeed.authorization.client.claims.IndeedAccessTokenClaimsSet.AZP_CLAIM_NAME;
+import static com.indeed.authorization.client.claims.IndeedAccessTokenClaimsSet.EXP_CLAIM_NAME;
+import static com.indeed.authorization.client.claims.IndeedAccessTokenClaimsSet.SCOPE_CLAIM_DELIMITER;
+import static com.indeed.authorization.client.claims.IndeedAccessTokenClaimsSet.SCOPE_CLAIM_NAME;
+
+public class IndeedAccessTokenClaimsVerifier
+        implements JWTClaimsSetVerifier<JWKSecurityContext>, ClockSkewAware {
+    private final Issuer expectedIssuer;
+    private final ClientID expectedClientID;
+    private final String[] expectedScopes;
+    private int maxClockSkew;
+
+    /**
+     * Creates a new access token claims verifier.
+     *
+     * @param issuer The expected ID token issuer. Must not be {@code null}.
+     * @param clientID The expected client ID. Must not be {@code null}.
+     * @param scopes The expected scopes. Must not be {@code null}.
+     * @param maxClockSkew The maximum acceptable clock skew (absolute value), in seconds. Must be
+     *     zero (no clock skew) or positive integer.
+     */
+    public IndeedAccessTokenClaimsVerifier(
+            final Issuer issuer,
```

Comment:
I don't think the expected issuer is necessary to be configurable for clients. In production environment, if should simply always be `https://secure.indeed.com` (you might want to double check if there's https or trailing slash by inspecting a real token)

Reply:
This is being used internally as well. So the issuer may be different for internal users.

Reply:
That makes sense. How about providing another constructor where issuer and jwk url are hardcoded (so that external clients don't need worry about figuring out these values), while preserving this constructor for internal usage?

Reply:
That makes sense. I have also create #5 to do the same thing for OAuthClients.